src/v/kafka/server/datalake_throttle_manager.cc

```diff
@@ -230,7 +231,7 @@ void datalake_throttle_manager::setup_metrics() {
     }
     namespace sm = ss::metrics;
     _metrics.add_group(
-      "kafka:datalake:throttle",
+      prometheus_sanitize::metrics_name("kafka:datalake:throttle"),
```

Comment:
what's the impact of not sanitizing this? how this was caught? do you think it is possible to catch these with some sort of generic (ducktape) test i.e. when querying metrics endpoint?